server/src/main/java/org/apache/druid/segment/realtime/firehose/HttpFirehoseFactory.java

```diff
@@ -81,7 +100,13 @@ public HttpFirehoseFactory(
   @Override
   protected InputStream openObjectStream(URI object) throws IOException
   {
-    return object.toURL().openConnection().getInputStream();
+    URLConnection urlConnection = object.toURL().openConnection();
+    if (!"".equals(httpAuthenticationUsername) && !"".equals(httpAuthenticationPassword)) {
```

Comment:
Please add this to another method `openObjectStream(URI object, long start)` too.

Reply:
Also, please add a unit test which checks that the request property is set properly. You probably need to extract a common part to a new method which opens and annotates a connection for testing.